code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -143,10 +142,6 @@ private void activateServices(final Service initialService, final List<Service>
          */
         try {
             for (Service service : services) {
-                if (service.getId().equals(initialService.getId())) {
```

Comment:
We just re-enabled something that was enabled before, and temporarily disabled :) Basically when reconcile is called for the service from heatlcheck or global host update, the activateService should trigger state transition activation for the service we are operating on. Otherwise the reconcile will be happening, and service state won't reflect it.